### PR TITLE
Staging wrangling

### PR DESCRIPTION
**Pull request by Bod9001**
### Changelog:
CL: [New] Adds ability to cut To copy and paste
CL: [Improvement] Improved Dial graphics
CL: [Improvement] Improves dev search performance
CL: [Improvement] Invisible syringe text
CL: [Balance] Slip over soap while walking
CL: [Fix] Cyborgs can hear local now
CL: [Fix] Fixes magboots plus slippy water equals a weird slip condition
CL: [Fix]  Fixes antagonists not showing up on the end round screen
CL: [Fix] fixes audio not looping on some things
CL: [Fix] Fixes stacked items losing their stack when cooked
CL: [Fix]  stops things in pipes getting teleported by quantum pads
CL: [Fix]  hopefully fixes the lighting flickering on shuttles

### CODE REVIEW

File: UnityProject/Assets/Scripts/MapSaver/MapSaver.cs [hunk 1]
```diff
@@ -381,6 +381,110 @@ public bool RemoveEmptys()
...
+						{
+							if (AllowedPoints == null)
+							{
+								TilesToRemove.Add(Tile);
+							}
+							else
+							{
+								if (AllowedPoints.Contains(Tile.LocalPosition))
+								{
+									TilesToRemove.Add(Tile);
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/MapSaver/MapSaver.cs:404]: mama mia, nesta mia

File: UnityProject/Assets/Scripts/MapSaver/MapSaver.cs [hunk 2]
```diff
@@ -381,6 +381,110 @@ public bool RemoveEmptys()
...
+					}
+					else
+					{
+						if (AllowedPoints.Contains(Object.LocalPosition))
+						{
+							_ = Despawn.ServerSingle(Object.gameObject);
+						}
+					}
+				}
+			}
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/MapSaver/MapSaver.cs:486]: this function is nearly 200 lines long

**Review by MaxIsJoe — APPROVED** [review]
honestly I got tired of trying to maintain any level of code quality in this project
I absolve my hands completely from when someone makes fun of our spegetti code here again, because after several PRs of me constantly repeating myself; I've realized that I'm shouting into the void and that nothing will change.

You've won, I'm approving this PR as is.

**Comment by Bod9001** [comment]
Like code quality is good but when it's stuff like function slightly too long function has too many if statements,it feels like sonar cube complaints \<insert gravity falls meme>

give me the proper stuff like,  this code is copied a few times, might be worth moving it into one function, 

or you're doing the same thing every time you access this one function Maybe put that code inside of the function/make a function for it

**Comment by AtnerNT** [comment]
> but when it's stuff like function slightly too long function has too many if statements,

Long functions make code harder to understand on review, which makes it annoying when future developers have to come in.
Breaking a function up into appropriate smaller functions makes it easier to maintain code.

File: UnityProject/Assets/Scripts/MapSaver/MapSaver.cs [hunk 3]
```diff
@@ -381,6 +381,110 @@ public bool RemoveEmptys()
 				FieldsToRefresh.Clear();
 			}
 
+			if (Cut)
```
Comment (AtnerNT) [inline UnityProject/Assets/Scripts/MapSaver/MapSaver.cs:384]: Like this here could be extracted to a CutRegion() function.
So on review its just:
if(Cut == true) CutRegion();

Far easier to understand

File: UnityProject/Assets/Scripts/MapSaver/MapSaver.cs [hunk 4]
```diff
@@ -381,6 +381,110 @@ public bool RemoveEmptys()
...
+				{
+					Objects = MetaTileMap.GetComponentsInChildren<RegisterTile>(true);
+#if UNITY_EDITOR
+					CommonManagerEditorOnly.Instance.CustomNetworkManagerPrefab.ForeverIDLookupSpawnablePrefabs.Clear();
+					CommonManagerEditorOnly.Instance.CustomNetworkManagerPrefab.SetUpSpawnablePrefabsForEverIDManual();
+					CustomNetworkManager.Instance = CommonManagerEditorOnly.Instance.CustomNetworkManagerPrefab;
+#endif
+				}
+
+				foreach (var Object in Objects)
```
Comment (AtnerNT) [inline UnityProject/Assets/Scripts/MapSaver/MapSaver.cs:472]: This hypothetical Cut function could then be further divided into a CutRegionTiles() and a CutRegionObjects() method.

**Review by AtnerNT — COMMENTED** [review]
Just one example here, but 200 line methods shouldnt be a thing. Splitting it into like 5-6 40ish line functions with informative names, makes the logic far easier to follow, and also allows those specific steps to be reused in other scripts if nessacary. 
(Like CutRegion() could be called from a delete tool, instead of only being part of the Save function)

File: UnityProject/Assets/Scripts/MapSaver/MapSaver.cs (hunk 4 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/MapSaver/MapSaver.cs:472]: wanted to integrated as part of the tile map stuff but I don't wan the destroyed tiles  dropping stuff messing up the save

**Pull request merged**